airflow/www/security.py

```diff
@@ -524,10 +524,8 @@ def _merge_perm(self, action_name: str, resource_name: str) -> None:
         resource = self.get_resource(resource_name)
         perm = None
         if action and resource:
-            perm = (
-                self.appbuilder.get_session.query(self.permission_model)
-                .filter_by(action=action, resource=resource)
-                .first()
+            perm = self.appbuilder.get_session.scalar(
+                select(self.permission_model).filter_by(action=action, resource=resource)
```

Comment:
apply `limit(1)`

Reply:
But limiting this one is probably a good idea

Reply:
made the change